core/store/src/lib.rs

```diff
@@ -270,6 +273,43 @@ pub fn create_store(path: &str) -> Arc<Store> {
     Arc::new(Store::new(db))
 }
 
+fn get_all_block_hash_to_outcomes(store: &Store, block_hash: &CryptoHash) -> HashSet<CryptoHash> {
+    match store.get_ser(ColOutcomesByBlockHash, block_hash.as_ref()) {
+        Ok(Some(hash_set)) => hash_set,
+        Ok(None) => HashSet::new(),
+        Err(e) => panic!("Can't read DB, {:?}", e),
+    }
+}
+
+pub fn fill_col_outcomes_by_hash(store: &Store) {
+    let mut store_update = store.store_update();
+    let outcomes: Vec<ExecutionOutcomeWithIdAndProof> = store
+        .iter(ColTransactionResult)
+        .map(|key| {
+            ExecutionOutcomeWithIdAndProof::try_from_slice(&key.1)
+                .expect("BorshDeserialize should not fail")
+        })
+        .collect();
+    let mut block_hash_to_outcomes: HashMap<CryptoHash, HashSet<CryptoHash>> = HashMap::new();
+    for outcome in outcomes {
+        match block_hash_to_outcomes.entry(outcome.block_hash) {
+            Entry::Occupied(mut entry) => {
+                entry.get_mut().insert(outcome.id().clone());
+            }
+            Entry::Vacant(entry) => {
+                let mut hash_set = get_all_block_hash_to_outcomes(store, &outcome.block_hash);
```

Comment:
Can we reuse the relevant method from ChainStoreAccess?

Reply:
We cannot due to the dependency here.